Source/WTF/wtf/PlatformEnableCocoa.h

```diff
@@ -868,3 +868,12 @@
 #if !defined(ENABLE_THREADED_ANIMATION_RESOLUTION)
 #define ENABLE_THREADED_ANIMATION_RESOLUTION 1
 #endif
+
+#if !defined(ENABLE_WEBGL_IN_WORKERS) \
+    && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000) \
+    || ((PLATFORM(IOS) || PLATFORM(MACCATALYST)) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 170000) \
+    || (PLATFORM(WATCHOS) && __WATCH_OS_VERSION_MIN_REQUIRED >= 100000) \
+    || (PLATFORM(APPLETV) && __TV_OS_VERSION_MIN_REQUIRED >= 170000))
+#define ENABLE_WEBGL_IN_WORKERS 1
```

Comment:
It's a little weird to create this compile time flag to comment out code a runtime check. Are you seeking to make it impossible to turn the flag on?

You might consider keeping the above change, discarding the change in OffscreenCanvas.cpp and modifying `UnifiedWebPreferences.yaml` as follows:

```
AllowWebGLInWorkers:
  type: bool
  status: stable
  humanReadableName: "Allow WebGL in Web Workers"
  condition: ENABLE(WEBGL)
  defaultValue:
    WebKitLegacy:
      "ENABLE(WEBGL_IN_WORKERS)": true
      default: false
    WebKit:
      "ENABLE(WEBGL_IN_WORKERS)": true
      default: false
    WebCore:
      "ENABLE(WEBGL_IN_WORKERS)": true
      default: false
```

Reply:
If you don't want people turning this flag on outside an Engineering build, you could change the `status` to `unstable` instead.

Reply:
You're absolutely right, that's a much better way to implement this.

I don't think it matters if people can enable it outside of Engineering builds, we just want to set what the default is for now.